diff --git a/runtime/platform/globals.h b/runtime/platform/globals.h
index 29bbb969..ea5a09ee 100644
--- a/runtime/platform/globals.h
+++ b/runtime/platform/globals.h
@@ -104,7 +104,7 @@
 // for more information on predefined macros:
 //   - http://msdn.microsoft.com/en-us/library/b0084kay.aspx
 //   - with gcc, run: "echo | gcc -E -dM -"
-#if defined(__ANDROID__)
+#if defined(__ANDROID__) && !defined(__TERMUX__)
 
 // Check for Android first, to determine its difference from Linux.
 #define DART_HOST_OS_ANDROID 1
