diff --git a/DEPS b/DEPS
index a7f5fd584b..49401f0e64 100644
--- a/DEPS
+++ b/DEPS
@@ -510,73 +510,8 @@ deps = {
   'src/third_party/dart/third_party/pkg/yaml_edit':
    Var('dart_git') + '/yaml_edit.git' + '@' + Var('dart_yaml_edit_rev'),
 
-  'src/third_party/dart/tools/sdks/dart-sdk':
-   {'dep_type': 'cipd', 'packages': [{'package': 'dart/dart-sdk/${{platform}}', 'version': 'version:3.2.0-150.0.dev'}]},
-
   # WARNING: end of dart dependencies list that is cleaned up automatically - see create_updated_flutter_deps.py.
 
-  # Prebuilt Dart SDK of the same revision as the Dart SDK source checkout
-  'src/flutter/prebuilts/linux-x64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/linux-amd64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "linux" and download_dart_sdk'
-  },
-  'src/flutter/prebuilts/linux-arm64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/linux-arm64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "linux" and download_dart_sdk'
-  },
-  'src/flutter/prebuilts/macos-x64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/mac-amd64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "mac" and download_dart_sdk'
-  },
-  'src/flutter/prebuilts/macos-arm64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/mac-arm64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "mac" and download_dart_sdk'
-  },
-  'src/flutter/prebuilts/windows-x64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/windows-amd64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "win" and download_dart_sdk'
-  },
-  'src/flutter/prebuilts/windows-arm64/dart-sdk': {
-    'packages': [
-      {
-        'package': 'flutter/dart-sdk/windows-arm64',
-        'version': 'git_revision:'+Var('dart_revision')
-      }
-    ],
-    'dep_type': 'cipd',
-    'condition': 'host_os == "win" and download_dart_sdk and not release_candidate'
-  },
-
   'src/third_party/colorama/src':
    Var('chromium_git') + '/external/colorama.git' + '@' + '799604a1041e9b3bc5d2789ecbd7e8db2e18e6b8',
 
@@ -773,87 +708,11 @@ deps = {
      'dep_type': 'cipd',
    },
 
-  'src/flutter/third_party/gn': {
-    'packages': [
-      {
-        'package': 'gn/gn/${{platform}}',
-        'version': 'git_revision:b79031308cc878488202beb99883ec1f2efd9a6d'
-      },
-    ],
-    'dep_type': 'cipd',
-  },
-  'src/flutter/third_party/ninja': {
-    'packages': [
-      {
-        'package': 'infra/3pp/tools/ninja/${{platform}}',
-        'version': 'version:2@1.11.1.chromium.4',
-      }
-    ],
-    'dep_type': 'cipd',
-  },
-
   'src/buildtools/emsdk': {
    'url': Var('skia_git') + '/external/github.com/emscripten-core/emsdk.git' + '@' + 'a896e3d066448b3530dbcaa48869fafefd738f57',
    'condition': 'download_emsdk',
   },
 
-  # Clang on mac and linux are expected to typically be the same revision.
-  # They are separated out so that the autoroller can more easily manage them.
-  'src/buildtools/mac-x64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/mac-amd64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "mac"', # On ARM64 Macs too because Goma doesn't support the host-arm64 toolchain.
-    'dep_type': 'cipd',
-  },
-
-  'src/buildtools/mac-arm64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/mac-arm64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "mac" and host_cpu == "arm64"',
-    'dep_type': 'cipd',
-  },
-
-  'src/buildtools/linux-x64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/linux-amd64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "linux" and host_cpu == "x64"',
-    'dep_type': 'cipd',
-  },
-
-  'src/buildtools/linux-arm64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/linux-arm64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'host_os == "linux" and host_cpu == "arm64"',
-    'dep_type': 'cipd',
-  },
-
-  'src/buildtools/windows-x64/clang': {
-    'packages': [
-      {
-        'package': 'fuchsia/third_party/clang/windows-amd64',
-        'version': Var('clang_version'),
-      }
-    ],
-    'condition': 'download_windows_deps',
-    'dep_type': 'cipd',
-  },
-
   # GOMA
   'src/buildtools/mac-x64/goma': {
     'packages': [
@@ -912,29 +771,6 @@ deps = {
     'dep_type': 'cipd',
   },
 
-  # Get the SDK from https://chrome-infra-packages.appspot.com/p/fuchsia/sdk/core at the 'latest' tag
-  # Get the toolchain from https://chrome-infra-packages.appspot.com/p/fuchsia/clang at the 'goma' tag
-   'src/fuchsia/sdk/mac': {
-     'packages': [
-       {
-        'package': 'fuchsia/sdk/core/mac-amd64',
-        'version': 'Qjz4zE4Oe0AO_7T3f-M9BFLMSvDVKjNM4m4C1xpSplsC'
-       }
-     ],
-     'condition': 'host_os == "mac" and not download_fuchsia_sdk',
-     'dep_type': 'cipd',
-   },
-   'src/fuchsia/sdk/linux': {
-     'packages': [
-       {
-        'package': 'fuchsia/sdk/core/linux-amd64',
-        'version': 'hMA99PoacaRZVXHgric86nJsfgQsM7O8Wn7ZDLc6Kp4C'
-       }
-     ],
-     'condition': 'host_os == "linux" and not download_fuchsia_sdk',
-     'dep_type': 'cipd',
-   },
-
   'src/third_party/impeller-cmake-example': {
      'url': Var('github_git') + '/bdero/impeller-cmake-example.git' + '@' + 'ad78e80741349dda91c8a17fd16919655406e7d8',
      'condition': 'download_impeller_cmake_example',
@@ -951,16 +787,6 @@ deps = {
     'condition': 'download_impeller_cmake_example and host_os == "mac"',
     'dep_type': 'cipd',
   },
-
-  'src/third_party/google_fonts_for_unit_tests': {
-      'packages': [
-        {
-          'package': 'flutter/flutter_font_fallbacks',
-          'version': '14d38cffa1a2c9d4380094957aa63aee811b613b0dc53fd909557f61f6a9068c'
-        }
-      ],
-      'dep_type': 'cipd',
-  }
 }
 
 recursedeps = [
