diff --git a/build/config/compiler/BUILD.gn b/build/config/compiler/BUILD.gn
index d38a4cf..be86508 100644
--- a/build/config/compiler/BUILD.gn
+++ b/build/config/compiler/BUILD.gn
@@ -352,10 +352,24 @@ config("compiler") {
   if (is_linux) {
     cflags += [ "-pthread" ]
     ldflags += [ "-pthread" ]
+    # termux
+    cflags += [
+      "-ffunction-sections",
+      "-funwind-tables",
+      "-fno-short-enums",
+      "-nostdinc++",
+      "-mno-outline-atomics",
+      ]
+    ldflags += [
+      "-Wl,--no-undefined",
+      "-Wl,--exclude-libs,ALL",
+      "-fuse-ld=lld",
+
+      # Enable identical code folding to reduce size.
+      "-Wl,--icf=all",
+    ]
 
     if (current_cpu == "arm64") {
-      cflags += [ "--target=aarch64-linux-gnu" ]
-      ldflags += [ "--target=aarch64-linux-gnu" ]
       cflags += [ "-DBORINGSSL_CLANG_SUPPORTS_DOT_ARCH" ]
     }
   }
@@ -523,9 +537,8 @@ config("runtime_library") {
     cflags_objcc += [ "-nostdinc++" ]
 
     # Unwind seemes to be in these libraries in Linux.
-    if (!is_linux) {
-      ldflags += [ "-nostdlib++" ]
-    }
+    ldflags += [ "-nostdlib++" ]
+
     include_dirs = [
       "//flutter/third_party/libcxx/include",
       "//flutter/third_party/libcxxabi/include",
